source/OptimaJet.DWKit.StarterApplication/Middleware/ErrorReportingJobFilter.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Linq;
+using Bugsnag;
+using Hangfire.Common;
+using Hangfire.States;
+
+namespace OptimaJet.DWKit.StarterApplication.Middleware
+{
+    // See: https://github.com/HangfireIO/Hangfire/issues/553
+    public class ErrorReportingJobFilter : JobFilterAttribute, IElectStateFilter
+    {
+        protected readonly IClient bugsnagClient;
+
+        public ErrorReportingJobFilter(IClient bugsnagClient)
+        {
+            this.bugsnagClient = bugsnagClient;
+        }
+
+        public void OnStateElection(ElectStateContext context)
+        {
+            // the way Hangfire works is retrying a job X times (10 by default), so this wont be called directly with a 
+            // failed state sometimes.
```

Comment:
sometimes? what's the situation?

Reply:
The comments are copied from the code at:
https://github.com/HangfireIO/Hangfire/issues/553

From what I can tell (http://docs.hangfire.io/en/latest/extensibility/using-job-filters.html):
>All processes are implemented with Chain-of-responsibility pattern and can be intercepted like with ASP.NET MVC Action Filters.

There are many different state transitions that will fire the JobFilter.  This condition correlates with Exception conditions.